src/utils/profileLink.ts

```diff
@@ -0,0 +1,134 @@
+/*
+* This program and the accompanying materials are made available under the terms of the *
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
+* https://www.eclipse.org/legal/epl-v20.html                                      *
+*                                                                                 *
+* SPDX-License-Identifier: EPL-2.0                                                *
+*                                                                                 *
+* Copyright Contributors to the Zowe Project.                                     *
+*                                                                                 *
+*/
+
+import * as readYaml from "js-yaml";
+import * as writeYaml from "yamljs";
+import * as vscode from "vscode";
+import { IProfileLoaded, Logger } from "@zowe/imperative";
+import * as path from "path";
+import * as fs from "fs";
+import { Profiles } from "../Profiles";
+import { ZoweTreeNode } from "../abstract/ZoweTreeNode";
+import { IZoweTreeNode } from "../api/IZoweTreeNode";
+import { getZoweDir } from "../utils";
+import * as nls from "vscode-nls";
+const localize = nls.config({ messageFormat: nls.MessageFormat.file })();
+
+const LINKS_FOLDER = "profile_links";
+const FILE_SUFFIX = ".yaml";
+
+export async function getLinkedProfile(node: IZoweTreeNode, type: string, logger?: Logger) {
+    try {
+        if (node instanceof ZoweTreeNode) {
+            return findLinkedProfile(await getProfile(node), type);
+        }
+        throw new Error(localize("profileLink.notTreeItem", "Tree Item is not a Zowe Explorer item."));
+    } catch (err) {
+        if (logger) {
+            logger.warn(err.message);
+        }
+        throw (err);
+    }
+}
+
+export function getProfile(node: vscode.TreeItem) {
+    if (node instanceof ZoweTreeNode) {
+        return (node as ZoweTreeNode).getProfile();
+    }
+    throw new Error(localize("profileLink.notTreeItem", "Tree Item is not a Zowe Explorer item."));
+}
+
+export async function linkProfileDialog(aProfile: IProfileLoaded) {
+    let chosenName;
+    let chosenType;
+    if (aProfile) {
+        const possibles = Profiles.getInstance().getAllTypes().filter( (value) => value !== aProfile.type);
```

Comment:
Going back to one of my previous comments (having duplicate types), I assume this is where the impact will be reflected?
Maybe it's just not possible to get duplicates and I missed something when looking at the `allTypes` array creation.

Reply:
I can't find a previous comment about duplicate types did you mean zosmf to zosmf?. No i wouldn't let you choose that.
Here I am just finding a list of possible alternative types to (Normally zosmf) I did think I maybe should have extended this to exclude types that had no profile instances defined. But I only thought it.

Reply:
Sorry, I must've deleted (or not added) my previous comment.

Here is the block of code I was referring to:
https://github.com/zowe/vscode-extension-for-zowe/blob/b8fb7f9e8ecb23ebebee0aa1cb7ee234df6c3375/src/Profiles.ts#L127-L131

So, if I have multiple ftp or tso profiles, will it be mad (i.e. show ftp more than once when selecting the types) because of the possible duplicates?

Reply:
Maybe it's fine since if `.configurations` only contains one element per type?

Reply:
I was being dumb, configurations are the same as what we see for profile creation, a list of profiles types available, no matter the number of profiles.

I should've done my research before complaining!